features/templates.feature

```diff
@@ -12,3 +47,36 @@ Feature: Templates
           | template | number |
           |  valid   |   no   |
           |  invalid |  some  |
+
+    @ignore
+    Scenario Outline: Merges template into user's config
+        Given an existing pipeline with job main
+        And an existing template called Test
+        And user only specifies template Test in job main
+        When user starts the pipeline
+        Then job main executes what is specified in template
+
+      @ignore
+      Scenario Outline: Job config takes precedence over template config
+          Given an existing pipeline with job main
+          And an existing template called Test
+          And user specifies template Test in job main
+          And template Test has environment {TEST: FOO}
```

Comment:
This is really implementation-specific. What about other user-defined settings? It's not described elsewhere, so what happens if the user specifies an `image`?

It may be better to describe this scenario more generally, and say that "a user has some settings defined" and verify that "those settings were preserved/persisted".